components/script/dom/range.rs

```diff
@@ -286,38 +292,38 @@ impl Range {
             .iter()
             .for_each(|s| s.queue_selectionchange_task());
     }
-}
 
-impl RangeMethods for Range {
-    // https://dom.spec.whatwg.org/#dom-range-startcontainer
-    fn StartContainer(&self) -> DomRoot<Node> {
-        self.start.node.get()
+    fn abstract_range(&self) -> &AbstractRange {
+        self.upcast::<AbstractRange>()
     }
 
-    // https://dom.spec.whatwg.org/#dom-range-startoffset
-    fn StartOffset(&self) -> u32 {
-        self.start.offset.get()
+    pub fn start_container(&self) -> DomRoot<Node> {
+        self.abstract_range().StartContainer()
```

Comment:
I added these functions, because they are called in selections.rs.
And also there are many calls like `self.abstract_range().start().node()` and `self.abstractrange.StartContainer()`, they are slightly different from  `self.abstract_range().StartContainer()`, should we try to clean up it?

Reply:
I think either way as fine as long as we refer to things consistently as much as possible.